ui/app/components/role-edit.js

```diff
@@ -15,6 +15,9 @@ import keys from 'core/utils/key-codes';
 const LIST_ROOT_ROUTE = 'vault.cluster.secrets.backend.list-root';
 const SHOW_ROUTE = 'vault.cluster.secrets.backend.show';
 
+/**
+ * @type Class
+ */
```

Comment:
Since we have `allowJs` defined in the `tsconfig` typescript attempts to build a declaration file for `js` files so they can imported into `ts` files. There were a bunch of errors that seem to be coming from classic ember components that use mixins and then are imported and extended by another component. Adding this jsdoc annotation makes the error go away and since these are old patterns that should eventually be updated I don't think it's worth adding declaration files for these components.